apps/google-analytics-4/frontend/src/components/main-app/SlugWarningDisplay/SlugWarningDisplay.spec.tsx

```diff
@@ -0,0 +1,66 @@
+import SlugWarningDisplay from './SlugWarningDisplay';
+import { render, screen } from '@testing-library/react';
+import { mockSdk } from '../../../../test/mocks';
+import { getContentTypeSpecificMsg } from '../constants/noteMessages';
+import * as useSidebarSlug from 'hooks/useSidebarSlug/useSidebarSlug';
+
+jest.mock('@contentful/react-apps-toolkit', () => ({
+  useSDK: () => mockSdk,
+}));
+
+const useSidebarSlugMock = {
+  reportSlug: '',
+  slugFieldValue: '',
+  isContentTypeWarning: true,
+};
+
+const { findByText } = screen;
+
+const { noSlugContentMsg, notPublishedMsg } = getContentTypeSpecificMsg('Category');
+
+describe('SlugWarningDisplay for the analytics app', () => {
+  it('mounts with correct msg', async () => {
+    jest.spyOn(useSidebarSlug, 'useSidebarSlug').mockImplementation(() => ({
+      slugFieldIsConfigured: false,
+      contentTypeHasSlugField: false,
+      isPublished: true,
+      ...useSidebarSlugMock,
+    }));
+
+    render(<SlugWarningDisplay slugFieldInfo={{ slugField: '', urlPrefix: '' }} />);
+
+    const warningMsg = await findByText(
+      "The Category content type has not been configured for use with this app. It must have a field of type short text and must be added to the list of content types in this app's configuration."
```

Comment:
Maybe import the warning message from constants as you've done with the other tests?

Reply:
ah yes when I had originally tried this, the tests failed only because of the line break from formatting. But now that I have reworded the message I am able to import 👍 thanks for pointing out!